### ISSUE TITLE
Weights positives and negatives differently in cross entropy

### ISSUE THREAD

**Issue by trimcrae** [Mamie/cs273bRegulatoryEvolution#11]
To compensate for class imbalance.

From Anna:

Following up on our Monday meeting, here is a link to my version of keras with the weighted losses: 
https://github.com/annashcherbina/keras

So you might care about 2 things: 
weighted binary cross-entropy loss. 
w0 & w1 are lists of task weights for positive (w1) and negative (w0) examples. 
so generally: 
w1_t = n/number of positives for task t 
w0_t = n/number of negatives for task t 

So you can calculate these lists for your tasks and  provide them to the loss function. 
https://github.com/annashcherbina/keras/blob/master/keras/losses.py#L63 

2. 
basset architecture example that makes use of the weighted binary cross-entropy loss.
https://github.com/annashcherbina/keras/tree/master/cs273b_examples
Note that the "init_weights" argument is None by default. If you want to use randomly-initialized weights, you can keep it set to None. If you want to initialize weights from the ENCODE file (`encode-roadmap.dnase_tf-chip.batch_256.params.npz` ) you can pass the path to this file to the 'init_weights' argument, and the weights will be loaded from this file.
